packages/router-plugin/tests/delete-nodes.test.ts

```diff
@@ -0,0 +1,76 @@
+import { readFile, readdir } from 'node:fs/promises'
+import path from 'node:path'
+import { describe, expect, it } from 'vitest'
+
+import { compileCodeSplitReferenceRoute } from '../src/core/code-splitter/compilers'
+import type { Config, DeletableNodes } from '../src/core/config'
+
+function getFrameworkDir(framework: string) {
+  const files = path.resolve(
+    __dirname,
+    `./delete-nodes/test-files/${framework}`,
+  )
+  const snapshots = path.resolve(
+    __dirname,
+    `./delete-nodes/snapshots/${framework}`,
+  )
+  return { files, snapshots }
+}
+
+const testGroups: Array<{
+  name: string
+  deleteNodes: Array<DeletableNodes> | undefined
+}> = [
+  {
+    deleteNodes: undefined,
+    name: '1-delete-nodes-undefined',
+  },
+  { deleteNodes: [], name: '2-delete-nodes-empty' },
+  {
+    deleteNodes: ['ssr'],
+    name: '3-delete-nodes-ssr',
+  },
+]
+
+const frameworks: Array<Config['target']> = ['react', 'solid']
```

Comment:
This should probably now become a shared variable between `code-splitter.test.ts` and `delete-nodes.test.ts` so that any additions in the future (of other frameworks) don't cause drift.